### PR TITLE
Migrate Cadence documentation

### PR DESCRIPTION
**Pull request by turbolent**
- Properly migrate the Cadence documentation by splicing in the Git commits relating to the documentation subdirectory from the Cadence repository, and re-apply the renaming and link changes. Effectively, this results in the same content as it is now, but preserves history and attribution.

- Remove documentation that is intended for *contributors* to Cadence, not *users* of Cadence. The deleted content, the contribution documentation, will stay in the Cadence repository.

### CODE REVIEW

File: docs/cadence/tutorial/index.md
Comment (turbolent) [inline docs/cadence/tutorial/index.md:1]: @nialexsan @fotescodev Why was this empty file added?

Comment (nialexsan) [reply docs/cadence/tutorial/index.md:1]: it was a placeholder for correct links resolution, but we can remove it if we don't need it

Comment (nialexsan) [reply docs/cadence/tutorial/index.md:1]: there are a few references to `cadence/tutorial` in the docs, and during the build time compiler checks these links and throws warnings like 
```
Frequent broken links are linking to:
- /cadence
- /cadence/tutorial
Exhaustive list of all broken links found:
- On source page path = /404.html:
   -> linking to /cadence
   -> linking to /cadence/tutorial
- On source page path = /community:
   -> linking to /cadence
   -> linking to /cadence/tutorial
- On source page path = /getting-started:
   -> linking to /cadence
   -> linking to /cadence/tutorial
...

```

Comment (turbolent) [reply docs/cadence/tutorial/index.md:1]: For /cadence: Can we make it redirect to /cadence/intro?
For /cadence/tutorial, @joshuahannan what would you like to happen there? For now I'll just restore the empty page

Comment (turbolent) [reply docs/cadence/tutorial/index.md:1]: Sounds good! Let's fix this in the follow-up PR: https://github.com/onflow/docs/pull/59

Comment (turbolent) [reply docs/cadence/tutorial/index.md:1]: @nialexsan Can you please advise on how to achieve this?

**Review by dsainati1 — APPROVED** [review]

**Pull request merged**